torch/_inductor/dependencies.py

```diff
@@ -255,6 +259,42 @@ def __init__(self, var_ranges: VarRanges, normalize: bool):
         super().__init__(parent_handler=parent_handler)
 
 
+class RecordListLoadStores(V.MockHandler):
+    def __init__(self):
+        super().__init__()
+        self._reads: Set[Dep] = set()
+        self._list_reads: Set[str] = set()
+        self._writes: Set[Dep] = set()
+        self._list_writes: Set[str] = set()
+
+    def load(self, list_name, _) -> str:
+        self._list_reads.add(list_name)
+        for buf_name in V.graph.lists[list_name]:
+            buffer = V.graph.get_buffer(buf_name)
+            (
+                _,
+                index,
+                _,
+            ) = index_vars_squeeze(buffer.layout.size)
+            indexer = buffer.layout.make_indexer()
+            self._reads.add(
+                MemoryDep(buf_name, indexer(*index), tuple(buffer.layout.size))
+            )
+
+    def store(self, name: str, layouts, value: str) -> None:
+        for buf_name in V.graph.lists[name]:
+            buffer = V.graph.get_buffer(buf_name)
+            (
+                _,
+                index,
+                _,
+            ) = index_vars_squeeze(buffer.layout.size)
+            indexer = buffer.layout.make_indexer()
+            self._writes.add(
+                MemoryDep(buf_name, indexer(*index), tuple(buffer.layout.size))
+            )
```

Comment:
Don't do layout -> indexing inside `ops.load()` and `ops.store()`.  That should happen in the callers.

Ideally `RecordListLoadStores` wouldn't exist and we could just reuse `RecordLoadStores`.  I don't see why this is needed.